packages/dev/core/src/Misc/tools.ts

```diff
@@ -416,15 +449,41 @@ export class Tools {
         });
     }
 
+    /**
+     * Get a script URL including preprocessing
+     * @param scriptUrl the script Url to process
+     * @returns a modified URL to use
+     */
+    public static GetScriptUrl(scriptUrl: Nullable<string>, forceAbsoluteUrl?: boolean): string {
+        if (!scriptUrl) {
+            return "";
+        }
+        // check if the URL is absolute or relative. Otherwise, append the base URL
+        if (!Tools.IsAbsoluteUrl(scriptUrl)) {
+            scriptUrl = Tools.ScriptBaseUrl + (scriptUrl[0] === "/" ? scriptUrl.substring(1) : scriptUrl);
+        }
+
+        // run the preprocessor
+        scriptUrl = Tools.ScriptPreprocessUrl(scriptUrl);
+
+        if (forceAbsoluteUrl) {
+            scriptUrl = Tools.GetAbsoluteUrl(scriptUrl);
+        }
+
+        return scriptUrl;
+    }
+
     /**
      * Load a script (identified by an url). When the url returns, the
      * content of this file is added into a new script element, attached to the DOM (body element)
-     * @param scriptUrl defines the url of the script to laod
+     * @param scriptUrl defines the url of the script to load
      * @param onSuccess defines the callback called when the script is loaded
      * @param onError defines the callback to call if an error occurs
      * @param scriptId defines the id of the script element
      */
     public static LoadScript(scriptUrl: string, onSuccess: () => void, onError?: (message?: string, exception?: any) => void, scriptId?: string) {
+        scriptUrl = Tools.GetScriptUrl(scriptUrl);

```

Comment:
This is a breaking change if we used it from outside babylon to load external scripts.

I think for those case we should add smthg like LoadBabylonScript ? as this only works for our dependencies ?

Reply:
Done :-)